oxidation/libparsec/crates/core_fs/src/storage/chunk_storage.rs

```diff
@@ -0,0 +1,405 @@
+
+    fn create_db(&self) -> FSResult<()> {
+        let conn = &mut *self.conn().lock().expect("Mutex is poisoned");
+        sql_query(
+            "CREATE TABLE IF NOT EXISTS chunks (
+                chunk_id BLOB PRIMARY KEY NOT NULL, -- UUID
+                size INTEGER NOT NULL,
+                offline BOOLEAN NOT NULL,
+                accessed_on REAL, -- Timestamp
+                data BLOB NOT NULL
+            );",
+        )
+        .execute(conn)
+        .map_err(|e| FSError::CreateTable(format!("chunks {e}")))?;
+        Ok(())
+    }
+
+    #[cfg(test)]
+    fn drop_db(&self) -> FSResult<()> {
+        let conn = &mut *self.conn().lock().expect("Mutex is poisoned");
+        sql_query("DROP TABLE IF EXISTS chunks;")
+            .execute(conn)
+            .map_err(|e| FSError::DropTable(format!("chunks {e}")))?;
+        Ok(())
+    }
+
+    // Size and chunks
+
+    fn get_nb_blocks(&self) -> FSResult<i64> {
+        let conn = &mut *self.conn().lock().expect("Mutex is poisoned");
+        chunks::table
+            .select(count_star())
+            .first(conn)
+            .map_err(|e| FSError::QueryTable(format!("chunks: get_nb_blocks {e}")))
+    }
+
+    fn get_total_size(&self) -> FSResult<i64> {
+        let conn = &mut *self.conn().lock().expect("Mutex is poisoned");
+        chunks::table
+            .select(CoalesceTotalSize::default())
+            .first(conn)
+            .map_err(|e| FSError::QueryTable(format!("chunks: get_total_size {e}")))
+    }
+
+    fn is_chunk(&self, chunk_id: ChunkID) -> FSResult<bool> {
+        let conn = &mut *self.conn().lock().expect("Mutex is poisoned");
+        chunks::table
+            .select(count_star())
+            .filter(chunks::chunk_id.eq((*chunk_id).as_ref()))
+            .first::<i64>(conn)
+            .map_err(|e| FSError::QueryTable(format!("chunks: is_chunk {e}")))
+            .map(|res| res > 0)
+    }
+
+    fn get_local_chunk_ids(&self, chunk_ids: &[ChunkID]) -> FSResult<Vec<ChunkID>> {
+        let bytes_id_list = chunk_ids
+            .iter()
+            .map(|chunk_id| (**chunk_id).as_ref())
+            .collect::<Vec<_>>();
+
+        let mut res = Vec::with_capacity(chunk_ids.len());
+
+        let conn = &mut *self.conn().lock().expect("Mutex is poisoned");
+        for bytes_id_list_chunk in bytes_id_list.chunks(SQLITE_MAX_VARIABLE_NUMBER) {
+            res.append(
+                &mut chunks::table
+                    .select(chunks::chunk_id)
+                    .filter(chunks::chunk_id.eq_any(bytes_id_list_chunk))
+                    .load::<Vec<u8>>(conn)
+                    .map_err(|e| FSError::QueryTable(format!("chunks: get_local_chunk_ids {e}")))?
+                    .into_iter()
+                    .map(|chunk_id| {
+                        <[u8; 16]>::try_from(&chunk_id[..]).map_err(|_| {
+                            FSError::QueryTable(format!("chunks: corrupted chunk_id {chunk_id:?}"))
```

Comment:
That wasn't what I mean, do we want to display the raw value of `chunk_id` ?

Reply:
After reading the previous answer, we want to keep the `chunk_id` in the error message